src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -419,6 +419,8 @@ public bool IsNavigableMetadataSymbol(ISymbol symbol)
             return false;
         }
 
+        public MetadataAsSourceWorkspace? GetWorkspace() => _workspace;
```

Comment:
I admit I don't like that this is exposing the laziness of the underlying workspace, but I don't have a better idea. 😢